<html><head><meta charset='utf-8'><meta name='viewport' content='width=device-width, initial-scale=1'>
<meta name='applicable-device' content='pc'><meta name='keywords' content='电脑,电脑讲解,电脑技术,编程,电脑故障维修Java自学-异常处理 Throwable' />
<script src='../../highlight/highlight.pack.js'></script>
<link rel='stylesheet' type='text/css' href='../../highlight/styles/monokai.css'/>

<link rel='stylesheet' href='../../fenxiang/dist/css/share.min.css'>
<script src='../../fenxiang/src/js/social-share.js'></script>
<script src='../../fenxiang/src/js/qrcode.js'></script>

</head><body><script>hljs.initHighlightingOnLoad();</script><script>
var system ={};  
var p = navigator.platform;       
system.win = p.indexOf('Win') == 0;  
system.mac = p.indexOf('Mac') == 0;  
system.x11 = (p == 'X11') || (p.indexOf('Linux') == 0);     
if(system.win||system.mac||system.xll){
document.write("<link href='../css/3.css' rel='stylesheet' type='text/css'>");}else{ document.write("<link href='../css/3wap.css' rel='stylesheet' type='text/css'>");}</script><script src='../../js/3.js'></script><div class='div2'><div class='heading_nav'><ul><div><li><a href='../../index.html'>首页</a></li>
</div><div onclick='hidden1()' >分享</div>
</ul></div></div>
<div id='heading_nav2'> 
<li class='row' >
<div class='social-share' data-mode='prepend'><a href='javascript:' class='social-share-icon icon-heart'></a></div></li></div><script charset='utf-8' src='../../3/js/hengfu.js'></script><script charset='utf-8' src='../../3/js/hengfu2.js'></script><hr><div class='div1'><div class='biaoti'><center>Java自学-异常处理 Throwable</center></div><div class='banquan'>原文出处:本文由博客园博主半米高峰提供。<br/>
原文连接:https://www.cnblogs.com/jeddzd/p/11684572.html</div><br>
    <h2 id="java-throwable类">Java Throwable类</h2>
<p>步骤 1 : <strong>Throwable</strong></p>
<p>Throwable是类，Exception和Error都继承了该类<br />
所以在捕捉的时候，也可以使用Throwable进行捕捉<br />
如图： 异常分<strong>Error</strong>和<strong>Exception</strong><br />
Exception里又分<strong>运行时异常</strong>和<strong>可查异常</strong>。<br />
<img src="./images/Java自学-异常处理 Throwable0.png" alt="Throwable" /></p>
<pre><code><code>package exception;
 
import java.io.File;
import java.io.FileInputStream;
 
public class TestException {
 
    public static void main(String[] args) {
 
        File f = new File(&quot;d:/LOL.exe&quot;);
 
        try {
            new FileInputStream(f);
            //使用Throwable进行异常捕捉
        } catch (Throwable t) {
            // TODO Auto-generated catch block
            t.printStackTrace();
        }
 
    }
}</code></code></pre>
<p><strong>练习</strong>：<a href="http://how2j.cn/k/exception/exception-throwable/334.html?p=43278">Throwable类</a></p>
<p>在方法声明上，可以抛出指定的异常，比如FileNotFoundException<br />
那么能否抛出Throwable这个类？</p>
<p>这个方法的调用者又该如何处理？</p>
<p><strong>答案</strong>：</p>
<p>可以抛出Throwable，并且在调用的时候必须进行catch处理。</p>
<p>但是这样的设计方法做不好，因为不知道抛出的类型到底是哪种具体问题，无法针对性的处理。</p>
<pre class="java"><code>package exception;
 
import java.io.File;
import java.io.FileInputStream;
 
public class TestException {
 
    public static void method() throws Throwable {
        File f = new File(&quot;d:/LOL.exe&quot;);
        new FileInputStream(f);
    }
 
    public static void main(String[] args) {
        try {
            method();
        } catch (Throwable e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
    }
}</code></code></pre>


</div><hr><script charset='utf-8' src='../../js/sming.js'></script></body></html>